criu/flog.c

```diff
@@ -150,12 +153,30 @@ int flog_close(int fdout)
 	return 0;
 }
 
-int flog_encode_msg(int fdout, unsigned int nargs, unsigned int mask, const char *format, ...)
+int flog_encode_msg(int loglevel, unsigned int nargs, unsigned int mask, const char *format, ...)
 {
 	flog_msg_t *m;
 	va_list argptr;
 	char *str_start, *p;
 	size_t i;
+	int fdout;
+	unsigned int current_loglevel;
+
+	if (!opts.log_in_binary){
+		va_start(argptr, format);
+		vprint_on_level(loglevel, format, argptr);
+		va_end(argptr);
+		return 0;
+	}
+
+	if (unlikely(loglevel == LOG_MSG)) {
+		fdout = STDOUT_FILENO;
+	}else {
+		current_loglevel = log_get_loglevel();
+		if (loglevel > current_loglevel)
+			return 0;
+		fdout = log_get_fd();
```

Comment:
The same question is about stderr: I see you print messages there unencoded (in flog.c), should you do the same for criu if the destination is stderr?

Reply:
I have been looking for ways to identify if it is printed in the stderr, but since the fd value that we get via the log_get_fd() doesn't match the standard stderr fd value, I am not able to differentiate between the one being printed in the file and one being printed on stdout/stderr. 
I am thinking of making the -b option available exclusively with the -o option by introducing variables in the parse_options() function. This variable will tell us if it's being logged in a file or not.

Reply:
> I am thinking of making the -b option available exclusively with the -o option

Sounds like a good idea to me. You can also try using `isatty()` as an additional check.